### PR TITLE
Various updates

### PR DESCRIPTION
**Pull request by alejoe91**
- make `center` preprocessor more efficient
- use waveforms and amp params and **kwargs in postprocessing
- improved and completed curation and validaiton tests
- fix bug in get_unit_waveforms https://github.com/SpikeInterface/spiketoolkit/issues/314
- unify param dicts between postprocessing and validation modules

### CODE REVIEW

File: spiketoolkit/preprocessing/center.py [hunk 1]
```diff
@@ -10,23 +10,41 @@ class CenterRecording(TransformRecording):
...
         else:
-            self._offset = -np.median(self._recording.get_traces(), axis=1)
+            self._offset = -np.median(traces, axis=1)
         TransformRecording.__init__(self, self._recording, scalar=self._scalar, offset=self._offset)
         self.copy_channel_properties(recording=self._recording)
         self._kwargs = {'recording': recording.make_serialized_dict(), 'mode': self._mode}
 
 
-def center(recording, mode='median'):
+def center(recording, mode='median', seconds=10, n_snippets=10):
```
Comment (colehurwitz) [inline spiketoolkit/preprocessing/center.py]: Maybe change seconds to 10.0 (not sure if this is needed or not though, your choice)

File: spiketoolkit/preprocessing/center.py [hunk 2]
```diff
@@ -36,9 +54,14 @@ def center(recording, mode='median'):
         The recording extractor to be transformed
     mode: str
         'median' (default) or 'mean'
+    seconds: float
+        Number of seconds used to compute center
+    n_snippets: int
+        Number of snippets in which the total 'seconds' are divided spannign the recording duration
```
Comment (colehurwitz) [inline spiketoolkit/preprocessing/center.py]: spanning*

File: spiketoolkit/tests/test_curation.py [hunk 3]
```diff
@@ -46,89 +47,200 @@ def test_thresh_num_spikes():
 
     assert np.all(new_ns >= s_threshold)
     check_dumping(sort_ns)
+    shutil.rmtree('test')
+
+
+def test_thresh_isi_violations():
```
Comment (colehurwitz) [inline spiketoolkit/tests/test_curation.py]: Awesome. These tests are great :)

File: spiketoolkit/tests/test_curation.py [hunk 4]
```diff
@@ -46,89 +47,200 @@ def test_thresh_num_spikes():
...
+    sort = create_dumpable_sorting(duration=10, num_channels=4, K=10, seed=0, folder='test')
```
Comment (alejoe91) [inline spiketoolkit/tests/test_curation.py]: move this to spikeextractors

**Pull request merged**